datacapturing/src/main/java/de/cyface/datacapturing/DataCapturingService.java

```diff
@@ -1110,8 +1111,19 @@ public void changeModalityType(@NonNull final Modality newModality) {
             // Record modality-switch Event for ongoing Measurements
             Log.v(TAG, "changeModalityType(): Logging Modality type change!");
             final Measurement measurement;
-
             measurement = loadCurrentlyCapturedMeasurement();
+
+            // Ensure the newModality is actually different to the current Modality
+            final List<Event> modalityChanges = persistenceLayer.loadEvents(measurement.getIdentifier(), Event.EventType.MODALITY_TYPE_CHANGE);
+            if (modalityChanges.size() > 0) {
```

Comment:
I would not know why. Current logic is:
- if there is already *a* modality shift, check if this is the same as the newModality which we are about to add - if so don' add it.

we don't want to check if the two last entries are the same - this would be too late (because than we already entered the duplicate shift).